client/src/components/edit-user-component/index.jsx

```diff
@@ -9,46 +11,56 @@ import {
   Typography,
 } from "@mui/material";
 import { Form, Header, Cancel } from "./index.styles";
+import ConfirmDialog from "../shared/confirm-dialog-component";
+import { modifyUser } from "../../utils/api";
+import { ErrorMessage } from "../login-component/index.styles";
+
+function EditComponent({ userID }) {
+  const navigate = useNavigate();
 
-function EditComponent() {
   const [firstName, setFirstName] = useState("");
   const [lastName, setLastName] = useState("");
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [confirmEditDialog, setConfirmEditDialog] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
 
   const handleSave = (e) => {
     e.preventDefault();
-    // Create a JSON object with the required keys and values
+    setConfirmEditDialog(true);
+  };
+
+  const confirmEdit = async () => {
+    setConfirmEditDialog(false);
     const userData = {
+      userID,
       firstName,
       lastName,
       email,
-      password,
+      isAdmin: false,
     };
 
-    // Convert the JSON object to a string
-    const userDataJSON = JSON.stringify(userData);
+    setIsLoading(true);
+    setErrorMessage("");
+    try {
+      const response = await modifyUser(userData);
+      if (response.ok) {
+        navigate("/admin");
+      } else {
+        const errorData = await response.json();
+        setErrorMessage(errorData.message || "User modification failed.");
+      }
+    } catch (error) {
+      setErrorMessage("An error occurred during your request.");
+    } finally {
+      setIsLoading(false);
+    }
+  };
 
-    // Replace url with target route
-    fetch("http://localhost:8000/edit", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: userDataJSON,
-    });
-    // .then((response) => {
-    //     if (response.ok) {
-    //         // Handle success response (e.g., redirect or show a success message)
-    //         console.log('Login successful');
-    //     } else {
-    //         // Handle error response (e.g., show an error message)
-    //         console.error('Login failed');
-    //     }
-    // })
-    // .catch((error) => {
-    //     console.error('Error:', error);
-    // });
+  const cancelEdit = () => {
+    setConfirmEditDialog(false);
   };
 
   return (
```

Comment:
Can we prepopulate the fields so that it is easier to make the edit to the user now that we have the users GET request added